dotnet/src/Skills/Skills.OpenAPI/Builders/Query/QueryStringBuilder.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Builders.Query;
+
+/// <summary>
+/// Represents a query string builder for REST API operations.
+/// </summary>
+internal class QueryStringBuilder : IQueryStringBuilder
+{
+    ///<inheritdoc/>
+    public string Build(RestApiOperation operation, IDictionary<string, string> arguments)
+    {
+        var queryStringSegments = new List<string>();
```

Comment:
```
NameValueCollection queryString = System.Web.HttpUtility.ParseQueryString(string.Empty);
queryString.Add("key1", "value1");
queryString.Add("key2", "value2");

return queryString.ToString(); // Returns "key1=value1&key2=value2", all URL-encoded
```

Reply:
It's refactoring; no functionality has changed; the code has just moved as it was:
* To simplify the code review of this PR.
* To establish the "builder" structure as preparation for the next PR, which will completely replace the code to support various query string serialization styles.
<img width="808" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/842d56d6-f0ab-4b30-aa75-b6b0181d4f7f">

The HttpUtility.ParseQueryString will be used in the next PR.